tests/integration/awslambda/test_lambda_integration.py

```diff
@@ -420,6 +420,139 @@ def verify_failure_received():
+            # Test exists filter
+            (
+                {"id": {"S": "test123"}},
+                {"id": {"S": "test1234"}, "presentKey": {"S": "test123"}},
+                {"dynamodb": {"NewImage": {"presentKey": [{"exists": False}]}}},
+                1,
+            ),
+            # numeric filters
+            (
+                {"id": {"S": "test123"}, "numericFilter": {"N": "123"}},
+                {"id": {"S": "test1234"}, "numericFilter": {"N": "12"}},
+                {"dynamodb": {"NewImage": {"numericFilter": [{"numeric": [">", 100]}]}}},
+                1,
+            ),
+            (
+                {"id": {"S": "test123"}, "numericFilter": {"N": "100"}},
+                {"id": {"S": "test1234"}, "numericFilter": {"N": "12"}},
+                {"dynamodb": {"NewImage": {"numericFilter": [{"numeric": [">=", 100, "<", 200]}]}}},
+                1,
+            ),
+            # Prefix
+            (
+                {"id": {"S": "test123"}, "prefix": {"S": "us-1-testtest"}},
+                {"id": {"S": "test1234"}, "prefix": {"S": "testtest"}},
+                {"dynamodb": {"NewImage": {"prefix": [{"prefix": "us-1"}]}}},
+                1,
+            ),
+        ],
+    )
+    def test_dynamo_db_insert_event(
+        self,
+        create_lambda_function,
+        lambda_client,
+        dynamodb_client,
+        dynamodb_create_table,
+        filter,
+        calls,
+        item_to_put1,
+        item_to_put2,
+    ):
+
+        function_name = f"lambda_func-{short_uid()}"
+        table_name = f"test-table-{short_uid()}"
+
+        try:
+            create_lambda_function(
+                handler_file=TEST_LAMBDA_PYTHON_ECHO,
+                func_name=function_name,
+                runtime=LAMBDA_RUNTIME_PYTHON37,
+            )
+            dynamodb_create_table(table_name=table_name, partition_key="id")
+            _await_dynamodb_table_active(dynamodb_client, table_name)
+            stream_arn = dynamodb_client.update_table(
+                TableName=table_name,
+                StreamSpecification={"StreamEnabled": True, "StreamViewType": "NEW_AND_OLD_IMAGES"},
+            )["TableDescription"]["LatestStreamArn"]
+            event_source_uuid = lambda_client.create_event_source_mapping(
+                FunctionName=function_name,
+                BatchSize=1,
+                StartingPosition="LATEST",
+                EventSourceArn=stream_arn,
+                MaximumBatchingWindowInSeconds=1,
+                MaximumRetryAttempts=1,
+                FilterCriteria={
+                    "Filters": [
+                        {"Pattern": json.dumps(filter)},
+                    ]
+                },
+            )["UUID"]
+            _await_event_source_mapping_enabled(lambda_client, event_source_uuid)
+            dynamodb_client.put_item(TableName=table_name, Item=item_to_put1)
+
+            def assert_single_lambda_call():
+                events = get_lambda_log_events(function_name)
```

Comment:
in order to use this helper function with AWS, we have to set the `logs_client`: `get_lambda_log_events(function_name, logs_client=logs_client)`